### PR TITLE
Feature task manager

### PR DESCRIPTION
**Pull request by diedpigs**

### CODE REVIEW

File: task_manager.py [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+            current['verify'][task_name] = success
+            routing_key = 'complete.' + username
+            done = success
+
+            logger.debug(f'Notify level task(s) done?{done}')
+
+    except Exception as exception:
+        logger.error('', exc_info=True)
+
+    if done:
```
Comment (rtripath89) [inline task_manager.py:99]: done=False variable needs to be defined outside if-else statements, else it gives the following error:
```
2020-04-24 17:07:50,009 [blocking_connection] - Created channel=1
Traceback (most recent call last):
  File "task_manager.py", line 126, in <module>
    'cb': task_manager
  File "/opt/rabbitmq_agents/rc_rmq.py", line 111, in start_consume
    self._channel.start_consuming()
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 1866, in start_consuming
    self._process_data_events(time_limit=None)
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 2027, in _process_data_events
    self.connection.process_data_events(time_limit=time_limit)
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 834, in process_data_events
    self._dispatch_channel_events()
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 566, in _dispatch_channel_events
    impl_channel._get_cookie()._dispatch_events()
  File "/usr/local/lib/python3.6/site-packages/pika/adapters/blocking_connection.py", line 1494, in _dispatch_events
    evt.properties, evt.body)
  File "task_manager.py", line 100, in task_manager
    if done:
UnboundLocalError: local variable 'done' referenced before assignment
```

**Review by rtripath89 — CHANGES_REQUESTED** [review]

File: task_manager.py (hunk 1 above)
Comment (diedpigs) [reply task_manager.py:99]: solved in 8a7f1f0

File: task_manager.py [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+args = rc_util.get_args()
+logger = rc_util.get_logger(args)
+
+record = {
+    'uid': -1,
+    'gid': -1,
+    'email': '',
+    'last_update': datetime.now(),
+    'request': {
+        'uid_resolve': None
```
Comment (eesaanatluri) [inline task_manager.py:19]: can you change the task name to `get_next_uid_gid` instead of `uid_resolve ` here.

Comment (diedpigs) [reply task_manager.py]: fixes in 39f3f08

**Pull request merged**